PySpice/Spice/HSpice/Server.py

```
@@ -0,0 +1,241 @@
+      with open(input_file, "w") as f:
+        f.write(netlist_str)
+
+      # Acquire the lock to limit the concurrency
+      lock_fd, slot_idx = acquire_hspice_lock(self._concurrency_limit)
+
+      try:
+        # Submit the job
+        # We run in the temporary directory to avoid cluttering current directory
+        cmd = [self._spice_command, "-i", "input.sp", "-o", "output"]
+        logger.info(f"Executing: {' '.join(cmd)}")
+        res = subprocess.run(cmd, cwd=tmp_dir, capture_output=True, text=True)
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)